routes.js

```diff
@@ -1,86 +1,86 @@
+      if(err) throw err;
+      console.log(newArray)
+    });
+    res.sendStatus(200);
+});
 
 
-// createMyCustomApi.put('/updateExistMember', (req, res) => {
-//   const {memberId, role , firstname, lastname} = req.body
-//   if(!memberId) {
-//     return res.status(400).send('Member ID is required');
-//   }
-//   const memberIndex = data.findIndex(member => member.memberId === memberId);
-//   if (memberId === -1) {
-//     return res.status(404).send("Member not found");
-//   }
-//   const updatedMember = {
-//     memberId,
-//     role : role,
-//     firstname : firstname ,
-//     lastname : lastname
-//   };
+createMyCustomApi.put('/updateExistMember', (req, res) => {
+  const {memberId, role , firstname, lastname} = req.body
+  if(!memberId) {
+    return res.status(400).send('Member ID is required');
+  }
+  const memberIndex = data.findIndex(member => member.memberId === memberId);
+  if (memberId === -1) {
+    return res.status(404).send("Member not found");
+  }
+  const updatedMember = {
+    memberId,
+    role : role,
+    firstname : firstname ,
+    lastname : lastname
+  };
 
-//   data[memberIndex] = updatedMember
+  data[memberIndex] = updatedMember
 
 
-//   saveData(data)
-//   res.status(201).send('Member updated successfully');
-// });
+  saveData(data)
+  res.status(201).send('Member updated successfully');
+});
 
 
 
-// let data = require('./members_Array.json'); // swsto datapath
+let data = require('./members_Array.json'); // swsto datapath
 
 
-// const saveData = (data) => {
-//   fs.writeFileSync('members_Array.json', JSON.stringify(data, null, 2)); // gia na grapsoume sto json arxeio
-// };
+const saveData = (data) => {
+  fs.writeFileSync('members_Array.json', JSON.stringify(data, null, 2)); // gia na grapsoume sto json arxeio
+};
 
-// createMyCustomApi.delete(`/removeMember/:memberId`, (req, res) => {
-//     const memberId = req.params.memberId;       
-//     const initialLength = data.length; // store initial length of our data
+createMyCustomApi.delete(`/removeMember/:memberId`, (req, res) => {
+    const memberId = req.params.memberId;       
+    const initialLength = data.length; // store initial length of our data
 
 
-//     data = data.filter(member => member.memberId !== memberId); // filtrarisma , create new array , xwris to member pou diagrafoume
+    data = data.filter(member => member.memberId !== memberId); // filtrarisma , create new array , xwris to member pou diagrafoume
 
 
-//     if(data.length < initialLength) {  // check ama mikrine to length diladi an diagraftike daga kai save
-//       saveData(data);
-//       res.status(204).send();
-//     }else {
-//       res.status(404).send('Member not found')
-//     }
+    if(data.length < initialLength) {  // check ama mikrine to length diladi an diagraftike daga kai save
+      saveData(data);
+      res.status(204).send();
+    }else {
+      res.status(404).send('Member not found')
+    }
```

Comment:
It is a good practice to keep the last line of a function a return of happy path. That way, anyone reading the function gets pretty good idea of what function does by just scanning it form top to bottom. Instead of trying to find function's intended result deep in nested if-statemnets
```suggestion
    if (data.length === initialLength) {
      res.status(404).send('Member not found')
    }
    saveData(data);
    res.status(204).send();
```
It will be also consistent with the validations you do in your other functions

Reply:
Done